OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -995,50 +959,47 @@ CPos ChooseRallyLocationNear(Actor producer)
 			return possibleRallyPoints.Random(Random);
 		}
 
-		void InitializeBase(Actor self)
+		void InitializeBase(Actor self, bool chooseLocation)
 		{
-			// Find and deploy our mcv
-			var mcv = self.World.Actors.FirstOrDefault(a => a.Owner == Player &&
-				Info.UnitsCommonNames.Mcv.Contains(a.Info.Name));
+			var mcv = FindAndDeployMcv(self, chooseLocation);
 
-			if (mcv != null)
-			{
-				initialBaseCenter = mcv.Location;
-				defenseCenter = mcv.Location;
-				QueueOrder(new Order("DeployTransform", mcv, false));
-			}
-			else
-				BotDebug("AI: Can't find BaseBuildUnit.");
+			if (mcv == null)
+				return;
+
+			initialBaseCenter = mcv.Location;
+			defenseCenter = mcv.Location;
 		}
 
-		// Find any newly constructed MCVs and deploy them at a sensible
-		// backup location.
-		void FindAndDeployBackupMcv(Actor self)
+		// Find any MCV and deploy them at a sensible location.
+		Actor FindAndDeployMcv(Actor self, bool move = true)
```

Comment:
You never rely on the fallback default of move, so there's no point including it.